.github/workflows/cmake.yml

```diff
@@ -0,0 +1,137 @@
+name: CMake
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    name: CMake with default option on the default GitHub Ubuntu
+    # The CMake configure and build commands are platform agnostic and should work equally
+    # well on Windows or Mac.  You can convert this to a matrix build if you need
+    # cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Install required packages
+        run: |
+          sudo apt update
+          sudo apt install libboost-all-dev librange-v3-dev
+
+      - name: Configure CMake
+        # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+        # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+      - name: Build
+        # Build your program with the given configuration
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        # Execute tests defined by the CMake configuration.
+        # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+        run: ctest -C ${{env.BUILD_TYPE}}
+
+  docker-gcc7-openmp:
+    name: OpenMP & OpenCL POCL with G++ 7
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - uses: docker://ubuntu:hirsute
+        env:
+          C_COMPILER: gcc-7
+          CXX_COMPILER: g++-7
+          OPENMP: ON
+          OPENCL: ON
+        with:
+          entrypoint: /github/workspace/.github/workflows/run_ci_in_docker.bash
+
+  docker-gcc11-openmp:
+    name: OpenMP with G++ 11
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - uses: docker://ubuntu:hirsute
+        env:
+          C_COMPILER: gcc-11
```

Comment:
You should be able to use `jobs.<job_id>.strategy` to reduce the amount of similar code here. But I'm not 100% sure that  you will like how it looks in the code, i.e. that is debatable. 

Names for jobs generated by `strategy` are auto-generated based on their arguments, including environment. For example, you can take a look at the SPIRV-LLVM-Translator repo: [link](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/actions/runs/1059720964/workflow#L40). Job names are visible on the left `strategy` directive should be highlighted in the workflow file

Reply:
Interesting... Up to now I have been trying just to have something working.
I am not yet to the level of optimizing.
It would be nice if the validation running inside the Docker image could be split in steps visible in the Action GUI instead of a single step "Run ubuntu:hirsute" but I do not know if it is easy to do...

Reply:
> Names for jobs generated by `strategy` are auto-generated based on their arguments, including environment.

I have added a specific `name` entry but it does appears only inside the job, on the Docker step. :-(
But I can survive to this.